### ISSUE TITLE
Compiler warning on returning from non-void function without return statement

### ISSUE THREAD

**Issue by lejcik** [Kistler-Group/sdbus-cpp#48]
Integration tests compilation emits the following warnings: (gcc 8.2.0)

```
| test/integrationtests/proxy-glue.h: In member function 'uint32_t testing_proxy::doOperationClientSideAsync(uint32_t)':
| test/integrationtests/proxy-glue.h:152:5: warning: no return statement in function returning non-void [-Wreturn-type]
|      }
|      ^
| test/integrationtests/proxy-glue.h: In member function 'uint32_t testing_proxy::doErroneousOperationClientSideAsync()':
| test/integrationtests/proxy-glue.h:160:5: warning: no return statement in function returning non-void [-Wreturn-type]
|      }
|      ^
```

What is the right approach for designing async methods? Should `doOperationClientSideAsync()` and `doErroneousOperationClientSideAsync()` be void functions?

BR,
Vilo

**Comment by sangelovic** [comment]
Thanks for reporting. This has been fixed in commit 99160156fe3c1355cb20251263b066c2f9cf910f.

**Comment by sangelovic** [comment]
Fixed in 99160156fe3c1355cb20251263b066c2f9cf910f
